### ISSUE TITLE
Make max_range compatible with NEP 34

### ISSUE THREAD

**Issue by Hoxbro** [holoviz/holoviews#4855]
#### Description of expected behavior and the observed behavior
When running a newer version of numpy it gives a warning when having a sequences-of-sequences where the lengths of the nested sequences do not match, see [NEP 34](https://numpy.org/neps/nep-0034-infer-dtype-is-object.html) for more information. 

I have observed this happens when `holoviews.core.util.max_range` is called with the minimal code example below, and comes from the following line.
https://github.com/holoviz/holoviews/blob/cab59419b419c2bfe8090fec4d6f9e0ff1bee45e/holoviews/core/util.py#L988

An example of values which give the warning is `values = [("data", "soft"), ("id", "data", "hard", "soft", "robust")]`.

A solution could be something like the following, but I feel like this is more avoiding the problem then fixing it.
``` python
dtype = object if len(set(map(len, values))) > 1 else None
arr = np.array(values, dtype=dtype)
```

I have not checked if there are other places in the code where a NEP 34 warning happens.

#### Complete, minimal, self-contained example code that reproduces the issue

``` python
import panel as pn
import hvplot.pandas  # noqa
from hvplot.sample_data import us_crime

crime = us_crime.read()
pn.panel(crime.hvplot.area(x="Year", y=["Robbery", "Aggravated assault"])).show()

```

**Comment by jlstevens** [comment]
Sounds like setting `dtype=object` would restore the old behavior but that probably isn't the correct fix either.

**Comment by philippjfr** [comment]
This was, afaik, fixed in 1.14.2.

**Comment by philippjfr** [comment]
Specifically, it was fixed here: https://github.com/holoviz/holoviews/pull/4809/files#diff-cae67649609b3488293d496c510d5c0eb1da92a86675a021aac3c8831d313282L123
